File: 03-Funciones/05-funcion-as-param/main.ts
// Executor function that takes a function as a parameter
function executor(a: number, b: number, local: (a: number, b: number) => number): number {
    return local(a, b);
}

// Addition function
function sumar(a: number, b: number): number {
    return a + b;
}

// Subtraction function
function restar(a: number, b: number): number {
    return a - b;
}

// Main execution
console.log("Función como parámetro");
console.log(`La suma es ${executor(8, 6, sumar)} y la resta es ${executor(8, 6, restar)}`);

File: 03-Funciones/05-funcion-as-param/main.go
package main

import "fmt"

func executor(a int, b int, local func(int, int) int) int {
	return local(a, b)
}

func sumar(a int, b int) int {
	result := a + b
	return result
}

func restar(a int, b int) int {
	result := a - b
	return result
}

func main() {
	fmt.Println("funcion como parametro")
	fmt.Printf("la suma es %d y la resta es %d \n", executor(8, 6, sumar), executor(8, 6, restar))
}
